fix: mobile display

mathiasayivor opened the pull request:
# Fixes Issue

**My PR closes #204 **

# 👨‍💻 Changes proposed(What did you do ?)
* Improvements on mobile display

# ✔️ Check List (Check all the applicable boxes)
<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->

<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

<!-- Add notes to reviewers if applicable -->

# 📷 Screenshots

<!-- Add all the screenshots which support your changes -->

Dun-sin commented:
Why is the changes so large? 🤔

mathiasayivor commented:
> Why is the changes so large? thinking

Most of the changes are class changes and the rest are lib function modifications.

Dun-sin commented:
> > Why is the changes so large? thinking
> 
> Most of the changes are class changes and the rest are lib function modifications.

Just for mobile responsiveness?

mathiasayivor commented:
> > > Why is the changes so large? thinking
> > 
> > 
> > Most of the changes are class changes and the rest are lib function modifications.
> 
> Just for mobile responsiveness?

Yep.

Dun-sin commented:
I don't think this request solves one thing, I'm seeing end search and some none other related things in there.

Can you break them into branches and issues, for me to easily review

mathiasayivor commented:
Aight, here's a breakdown of all the changes I made:
* I added @rsuite/icons library for Anonymous.jsx menu buttons (lines 85-95, 110-125, 134) 
* Moved `isSearching` && `currentChatId` status to context in order to use it for hiding the navbar when chatting with a user:
  Before:
  ![Screen Shot 2022-10-26 at 06 59 27](https://user-images.githubusercontent.com/74247733/197957404-38e89c77-1c48-4c2d-aa26-97b9255efcc4.png)
  ![Screen Shot 2022-10-26 at 06 58 29](https://user-images.githubusercontent.com/74247733/197957386-e5e490b6-dea4-43d1-bb5d-c38365ba2b75.png)

  After:
  ![Screen Shot 2022-10-26 at 07 00 01](https://user-images.githubusercontent.com/74247733/197957576-068c205b-e368-43eb-a7d4-112a1653f3e8.png)
  ![Screen Shot 2022-10-26 at 06 57 41](https://user-images.githubusercontent.com/74247733/197957588-fa4d3999-807e-4b8e-9224-24d7f8590650.png)

* Used the `createClassesFromArray` util to help add multiple classes in a readable manner (vue style)
* Updated loadingText to be properly displayed on mobile devices (BuddyMatcher.jsx line 36-51)
* The rest is just CSS class updates

Dun-sin commented on client/src/lib/utils.js:
```diff
@@ -1,8 +1,10 @@
+import { flattenDeep, uniq } from 'lodash';
+
 /**
  *
  * @param {any[]} classes
  * @returns
  */
-export function createClassesFromArray(classes) {
-    return classes.filter(Boolean).join(' ');
+export function createClassesFromArray(...classes) {
+    return uniq(flattenDeep(classes)).filter(Boolean).join(' ');
 }
```
What does this do exactly

mathiasayivor replied:
For adding multiple css classes easily in a readable and more maintainable way

mathiasayivor commented:
> I'm seeing end search and some none other related things in there.

@Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?

Dun-sin commented:
> > I'm seeing end search and some none other related things in there.
> 
> @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?

that's all i could notice

mathiasayivor commented:
> > > I'm seeing end search and some none other related things in there.
> > 
> > 
> > @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?
> 
> that's all i could notice

Aight.
But is my explanation clear enough?

Dun-sin commented:
> > > > I'm seeing end search and some none other related things in there.
> > > 
> > > 
> > > @Dun-sin besides `endSearch` and the changes I've explained above, what other none related things do you see?
> > 
> > 
> > that's all i could notice
> 
> Aight. But is my explanation clear enough?

yes it's clear, thanks
